### ISSUE TITLE
isReact doesn't work with hooks?

### ISSUE THREAD

**Issue by braco** [planttheidea/moize#139]
```
const TestComponent = moize(() => {
  const [txt, setTxt] = useState('before');
  useEffect(() => {
    setTimeout(() => {
      setTxt(Date.now());
    }, 1000);
  }, []);
  return <span>{txt}</span>;
}, {
  isReact: true,
});

```
Causes this error:

```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
```

`react@16.13.1` and `moize@6.0.0-beta.4`

**Comment by planttheidea** [comment]
Hmm ... this is interesting. I think I know why this may be happening, but I'll need to take a look. I'll keep you posted!

**Comment by planttheidea** [comment]
So I think I have a workaround for it, at least I was able to get hooks working with the changes made. I'll take a step back and check this out with fresh eyes later, but assuming it still looks good I'll merge and release beta 5.

**Comment by planttheidea** [comment]
The fix for this should have been published in `6.0.0-beta.5`. Let me know if you have any more issues!
